File: 2-even-fibonacci-numbers/2.py
# Find the sum of all even fibonacci numbers below 4000000

a = 1
b = 1
c = 2
total = 0

while c < 4000000:
    if c % 2 == 0:
        total += c
    a = b
    b = c
    c = a + b

print total 

File: haskell/2.hs
-- find sum of even fibonacci numbers less than 4 million

fibonacci :: Int -> [Int]
fibonacci max = listTo max [1,1] where
  listTo max (a:b:list) = if (a + b >= max)
                             then a:b:list
                             else listTo max ((a + b):a:b:list)
                                  
-- sum (filter even (fibonacci 4000000)) = 4613732